scripts/compile-triton.sh

```diff
@@ -33,18 +23,57 @@ for arg in "$@"; do
       VENV=true
       shift
       ;;
-    --skip-triton)
-      SKIP_TRITON=true
+    --help)
+      echo "Example usage: ./compile-triton.sh [--llvm | --triton | --clean | --venv]"
+      exit 1
+      ;;
+    *)
+      ARGS+="${arg} "
       shift
       ;;
   esac
 done
 
+if [ "$BUILD_LLVM" = false ] && [ "$BUILD_TRITON" = false ]; then
+  BUILD_LLVM=true
+  BUILD_TRITON=true
+fi
+
+set +o xtrace
+if [ -z "$BASE" ]; then
+  echo "**** BASE is not given *****"
+  BASE=$(cd $(dirname "$0")/../.. && pwd)
+  echo "**** Default BASE is set to $BASE ****"
+fi
+
+export PACKAGES_DIR=$BASE/packages
+export SPIRV_TOOLS=$PACKAGES_DIR/spirv-tools
+export LLVM_PROJ=$BASE/llvm
+export TRITON_PROJ=$BASE/intel-xpu-backend-for-triton
+export TRITON_PROJ_BUILD=$TRITON_PROJ/python/build
+
 if [ "$CLEAN" = true ]; then
-  echo "**** Cleaning $PACKAGES_DIR , $LLVM_PROJ , and $TRITON_PROJ_BUILD before build ****"
+  echo "**** Cleaning $PACKAGES_DIR, $LLVM_PROJ, and $TRITON_PROJ_BUILD before build ****"
   rm -rf $PACKAGES_DIR $LLVM_PROJ $TRITON_PROJ_BUILD
 fi
 
+if [ "$VENV" = true ]; then
+  echo "**** Creating Python virtualenv ****"
+  python3 -m venv .venv --prompt triton
+  source .venv/bin/activate
+  pip install ninja cmake wheel
+else
+  echo "**** Cleaning up Python virtualenv ****"
+  deactivate
+fi
```

Comment:
I know it was in the previous version, but `deactivate` generates an error if we are not in venv. The error is ignored, because the script does not have `set -e` but anyways it is an error.
```suggestion
elif [ -n "$VIRTUAL_ENV" ]; then
  echo "**** Cleaning up Python virtualenv ****"
  deactivate
fi
```